### PR TITLE
#155 refactor firestore documents to use sub collections

### PR DESCRIPTION
**Pull request by scratchclaggy**
## Change Summary

This PR represents major breaking changes, and the addition of two new JS libraries. The changes are:

- Remove the Firestore / Contact contexts
- Use [React Query](https://tanstack.com/query/v4/docs/react/overview) for all datafetching from Firestore
- Expose data fetching via hooks (see the `src/hooks` directory)
- Share state accross multiple components with [Jotai](https://jotai.org/docs/introduction)
- Change the user 'schema' in Firestore, so that visits and contacts are sub-collections, and each visit / contact is a separate document within their respective collection

### Change Form

- [x] The pull request title has an issue number
- [x] The change works by "Smoke testing" or quick testing
- [ ] The change has tests
- [x] The change has documentation

### CODE REVIEW

**Comment by scratchclaggy** [comment]
The WIP flag is so that I complete documentation before this is merged in.

**Comment by scratchclaggy** [comment]
Documentation is now included in the README.

File: src/hooks/user.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      email: currentUser.email ?? '',
+      phone: currentUser.phoneNumber ?? '',
+      streetAddress: '',
+      region: [],
+      pets: '',
+      tags: ['Volunteer'],
+      desc: '',
+      notes: ''
+    },
+    visits: []
```
Comment (NicholasChoong) [inline src/hooks/user.ts:31]: I dont think this array is needed

**Review by NicholasChoong — CHANGES_REQUESTED** [review]
Good work!! I read some of the documentations on Jotai and that they also have a library that integrates jotai into React Query. Surprise to see that Provider is optional in Jotai.

File: src/types/types.ts [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import { Timestamp } from 'firebase/firestore'
 
-export interface UserData {
+export interface User {
   info: Contact
-  contacts: Contact[]
-  visits: VisitData[]
+  visits: Visit[]
```
[the remark this answers is not in the document]
Comment (scratchclaggy) [reply src/types/types.ts:5]: Yeps, that's correct, thanks for picking that up.

**Review by NicholasChoong — APPROVED** [review]
Good to merge

**Pull request merged**